src/components/artwork_details.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from "react"
+import * as Relay from "react-relay"
+import styled from "styled-components"
+
+export class ArtworkDetails extends React.Component<RelayProps, null> {
+
+  artistLine() {
+    const cultural_maker = this.props.artwork.cultural_maker
+    const artists = this.props.artwork.artists
+
+    if (cultural_maker) {
+      return (<h4>{cultural_maker}</h4>)
+    } else if (artists && artists.length) {
+      let artistsEl = []
+      for (var i=0; i < artists.length; i++) {
+        artistsEl.push(<a href={artists[i].href} >{artists[i].name}</a>)
+        if (i !== artists.length - 1) {
+          artistsEl.push(<span>,</span>)
+        }
+      } 
+      return (<h4>{artistsEl}</h4>)
+    }
+  }
+
+  titleLine() {
+    return (
+      <a href={this.props.artwork.href}>
+        <em>{this.props.artwork.title}</em>
+        { this.props.artwork.date ? `, ${this.props.artwork.date}` : null }
+      </a>
+    )
+  }
+
+  partnerLine() {
+    if (this.props.artwork.collecting_institution) {
+      return (<div>{this.props.artwork.collecting_institution}</div>)
+    } else {
+      return (
+        <div>
+          <a href={this.props.artwork.partner.href}>
+            {this.props.artwork.partner.name}
+          </a>
+        </div>
+      )
+    }
+  }
+
+  saleLine() {
+    const artwork = this.props.artwork
+    const hasSaleMessage = artwork.sale_message && artwork.sale_message != 'Contact For Price'
+    const notInAuction = !(artwork.sale && artwork.sale.is_auction)
+    if (hasSaleMessage && notInAuction) {
+      return (
+        <div>{artwork.sale_message}</div>
+      )
+    }
+  }
+
+  render() {
+    return (
+      <div>
+        {this.artistLine()}
```

Comment:
Should each of these actually be it's own React component? I don't know how deep boxes on boxes goes (or should go)...